### PR TITLE
Require secp256k1 pubkey registration for unjail

### PR DESCRIPTION
**Pull request by hacheigriega**
## Explanation of Changes

We override the slashing module's unjail transaction to add an additional check for registration of required SEDA public keys. 

## Testing

Tested on a 4-node local network where one of the nodes was missing the required public key. When the proving scheme was activated, the node was jailed. Subsequently, its unjail attempts were blocked until it registered the required public key. 
```
sedad tx slashing unjail --from validator4 --keyring-backend test --home $HOME/.sedad/validator4 --keyring-dir $HOME/.sedad/validator4 --gas-prices 10000000000aseda --gas auto --gas-adjustment 2.0 --keyring-backend test --chain-id testing --node http://localhost:26657 --yes

Error: rpc error: code = Unknown desc = rpc error: code = Unknown desc = failed to execute message; message index: 0: validator has not registered required SEDA keys: invalid request [cosmossdk.io/errors@v1.0.1/errors.go:151] with gas used: '33552': unknown request
10:47AM ERR failure when running app err="rpc error: code = Unknown desc = rpc error: code = Unknown desc = failed to execute message; message index: 0: validator has not registered required SEDA keys: invalid request [cosmossdk.io/errors@v1.0.1/errors.go:151] with gas used: '33552': unknown request"
```

Note there is a bug where the keyring directory doesn't default to the home directory given by the `--home` flag, so the `--keyring-dir` flag has to be provided. Created issue #435

## Related PRs and Issues

Closes: #422

### CODE REVIEW

File: scripts/local_multi_setup.sh [hunk 1]
```diff
@@ -191,28 +191,11 @@ cat << EOF > validator4.json
...
-tmux send -t validator2 'C-c'
-tmux send -t validator2 '$BIN start --home=$HOME/.sedad/validator2 --log_level debug > val2_multi_local.log 2>&1 &' ENTER
-
-tmux send -t validator3 'C-c'
-tmux send -t validator3 '$BIN start --home=$HOME/.sedad/validator3 --log_level debug > val3_multi_local.log 2>&1 &' ENTER
-
-tmux send -t validator4 'C-c'
-tmux send -t validator4 '$BIN start --home=$HOME/.sedad/validator4 --log_level debug > val4_multi_local.log 2>&1 &' ENTER
+# val4 generates and uploads SEDA keys
+# $BIN tx pubkey add-seda-keys --from validator4 --keyring-backend test --home $HOME/.sedad/validator4 --gas-prices 10000000000aseda --gas auto --gas-adjustment 2.0 --keyring-backend test --chain-id=testing --node http://localhost:26657 --yes
```
Comment (Thomasvdam) [inline scripts/local_multi_setup.sh:201]: Do we want to leave this in?

File: x/slashing/msg_server.go [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+		MsgServer:             sdkMsgServer,
+		pubKeyKeeper:          pubKeyKeeper,
+		validatorAddressCodec: valAddrCdc,
+	}
+	return ms
+}
+
+// Unjail overrides the default Unjail method to add an additional
+// check for registration of required public keys.
+func (m msgServer) Unjail(ctx context.Context, req *types.MsgUnjail) (*types.MsgUnjailResponse, error) {
```
Comment (Thomasvdam) [inline x/slashing/msg_server.go:38]: Should this not also check whether the scheme has been activated? I see a point in not having that check, will get more validators to register it ahead of time :D

Comment (hacheigriega) [reply x/slashing/msg_server.go:39]: I didn't intend that 😂 Added the check

**Review by Thomasvdam — APPROVED** [review]

File: scripts/local_multi_setup.sh (hunk 1 above)
Comment (hacheigriega) [reply scripts/local_multi_setup.sh:201]: Yeah I thought we could use this comment to add val4's pubkey separately

**Pull request merged**